gocourse04/smartzoo/main_test.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"reflect"
+	"testing"
+)
+
+var lookupAreas = map[string]Area{
+	"Hoofed": {
+		Name: "hoofed", Sectors: map[string]Sector{
+			"Horses": {
+				ID: 1,
+				Animals: []Animal{
+					{ID: 1, Name: "Horse1"},
+					{ID: 2, Name: "Horse2"},
+				},
+			},
+		},
+	},
+	"Birds": {
+		Name: "birds", Sectors: map[string]Sector{
+			"Doves": {
+				ID: 3,
+				Animals: []Animal{
+					{ID: 3, Name: "Dove1"},
+					{ID: 4, Name: "Dove2"},
+				},
+			},
+		},
+	},
+	"Techroom": {
+		Name: "tech room", Sectors: map[string]Sector{
+			"Dining room": {
+				ID: 5,
+				Animals: []Animal{
+					{ID: 5, Name: "Hawk4"},
+				},
+			},
+			"Bath room": {
+				ID: 6,
+				Animals: []Animal{
+					{ID: 6, Name: "Cow4"},
+				},
+			},
+		},
+	},
+}
+
+var testsCleaning = map[string]Area{
+	"Hoofed": {
+		Name: "hoofed", Sectors: map[string]Sector{
+			"Horses": {
+				ID: 1,
+				Animals: []Animal{
+					{ID: 2, Name: "Horse2"},
+					{ID: 1, Name: "Horse1"},
+				},
+			},
+		},
+	},
+	"Birds": {
+		Name: "birds", Sectors: map[string]Sector{
+			"Doves": {
+				ID: 3,
+				Animals: []Animal{
+					{ID: 3, Name: "Dove1"},
+					{ID: 4, Name: "Dove2"},
+				},
+			},
+		},
+	},
+	"Techroom": {
+		Name: "tech room", Sectors: map[string]Sector{
+			"Dining room": {
+				ID: 5,
+				Animals: []Animal{
+					{ID: 5, Name: "Hawk4"},
+				},
+			},
+			"Bath room": {
+				ID: 6,
+				Animals: []Animal{
+					{ID: 6, Name: "Cow4"},
+				},
+			},
+		},
+	},
+}
+
+var testsMigrated = map[string]Area{
+	"Hoofed": {
+		Name: "hoofed", Sectors: map[string]Sector{
+			"Horses": {
+				ID: 1,
+				Animals: []Animal{
+					{ID: 2, Name: "Horse2"},
+				},
+			},
+		},
+	},
+	"Birds": {
+		Name: "birds", Sectors: map[string]Sector{
+			"Doves": {
+				ID: 3,
+				Animals: []Animal{
+					{ID: 3, Name: "Dove1"},
+					{ID: 4, Name: "Dove2"},
+				},
+			},
+		},
+	},
+	"Techroom": {
+		Name: "tech room", Sectors: map[string]Sector{
+			"Dining room": {
+				ID: 5,
+				Animals: []Animal{
+					{ID: 5, Name: "Hawk4"},
+					{ID: 1, Name: "Horse1"},
+				},
+			},
+			"Bath room": {
+				ID: 6,
+				Animals: []Animal{
+					{ID: 6, Name: "Cow4"},
+				},
+			},
+		},
+	},
+}
+
+func TestLookup(t *testing.T) {
+	var (
+		zoo      = &Zoo{Areas: lookupAreas}
+		wantName = "Horse1"
+	)
+
+	_, _, animal := zoo.Lookup(wantName)
+	if animal != nil && animal.Name != wantName {
+		t.Errorf("expected an animal named %s, but got %s", wantName, animal.Name)
+	}
+}
+
+func TestClean(t *testing.T) {
+	var (
+		zoo      = &Zoo{Areas: lookupAreas}
+		cleaning = &Zoo{Areas: testsCleaning}
+		name     = "Horse1"
+	)
+
+	AnimalArea, AnimalSector, Animal := zoo.Lookup(name)
+	Animal.Clean(AnimalArea.Name, AnimalSector.ID, *zoo)
```

Comment:
Локальні змінні завжди lowercase:
```suggestion
	area, sector, animal := zoo.Lookup(name)
	animal.Clean(area.Name, sector.ID, *zoo)
```

Reply:
Бажано виправити тут та нижче